packages/backend/src/open_payments/receiver/service.ts

```diff
@@ -179,3 +188,49 @@ async function getLocalIncomingPayment({
 
   return incomingPayment.toOpenPaymentsType({ ilpStreamConnection: connection })
 }
+
+async function getIncomingPaymentGrant(
+  deps: ServiceDependencies,
+  paymentPointerUrl: string
+): Promise<Grant | undefined> {
+  const paymentPointer = await deps.openPaymentsClient.paymentPointer.get({
+    url: paymentPointerUrl
+  })
+  if (!paymentPointer) {
+    return undefined
+  }
+  const grantOptions = {
+    authServer: paymentPointer.authServer,
+    accessType: AccessType.IncomingPayment,
+    accessActions: [AccessAction.ReadAll]
```

Comment:
I guess it all depends on the AS, but would just `Read` be "safer" here? Having to create a grant per incoming payment isn't great, but makes it easier on auth servers to restrict access

Reply:
I don't have a strong opinion either way, and I'm unsure which would make auth server locking more of a necessity:
https://github.com/interledger/rafiki/pull/779#discussion_r1029968121

Reply:
I think just `Read` wouldn't work because I'd imagine in most cases, this Rafiki instance wasn't the client creating the incoming payment. It just wants to read it to pay it.